e2e/page-objects/mapPage.ts

```
@@ -105,14 +146,18 @@ export class MapCanvas extends HelperBase {
    */
 
+  @step(
+    (args) =>
+      `Get viewport from application url ${args[0]?.url || ''} and check that map data is defined and that zoom, latitude and longitude are not NaN`,
+  )
   async getViewportFromUrl(page: Page = this.page) {
     const mapData = page.url().split('map=')[1].split('&')[0];
-    expect(mapData).toBeDefined();
+    expect(mapData, `Check that map data (${mapData}) is defined`).toBeDefined();
 
     const [zoom, latitude, longitude] = mapData!.split('/').map(Number);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)